src/pydicom/encaps.py

```diff
@@ -792,6 +792,8 @@ def encapsulate_extended(frames: list[bytes]) -> tuple[bytes, bytes, bytes]:
     """
     nr_frames = len(frames)
     frame_lengths = [len(frame) for frame in frames]
+    # Odd-length frames get padded to even length by `encapsulate()`
+    frame_lengths = [ii + 1 if ii % 2 else ii for ii in frame_lengths]
```

Comment:
Yuo could just use `[ii + ii % 2 for ii in frame_lengths]` - not that it would make a difference...